streams/src/main/java/org/apache/kafka/streams/processor/internals/PartitionGroup.java

```diff
@@ -84,6 +84,17 @@ RecordQueue queue() {
         streamTime = RecordQueue.UNKNOWN;
     }
 
+    long getPartitionTimestamp(final TopicPartition partition) {
```

Comment:
Should we check for partition existence for 2 calls?

Reply:
What do you mean by "for 2 calls"? The call to `partitionTime()`? If the partition does not exists, it's a bug anyway and current code throws a NPE. What do we gain by checking if the partition exists? There is still a bug and we would need to throw an exception, too, IMHO. What do we gain by manually throwing an exception?